### PR TITLE
Feat: Support Capacitor 6

### PR DESCRIPTION
**Pull request by lucas-zimerman**
Fixes: #630

This PR brings support to Capacitor while also adding some quality of life improvements. During the review, the ionic-angular-v2 was removed due to being a too old dependency of Capacitor that no longer works on the latest Android Studio, I opted to remove it since we already will cover a wide range of versions (3.0 up to 6.0). Also ionic-angular was renamed to ionic-angular-v4 to match the pattern where the last number indicates the capacitor version used by the sample.

Minor changes:

Samples:
- Added Capacitor 6 and 3 Samples.
- Removed Capcitor 2 Sample.
- Added script to skip pod install on Windows.

SDK:
- Add Required changes to support Capacitor 6.
- Code cleanup on gradle script in order to reduce dependencies and make it more simple to maintain.
-

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -1,60 +1,58 @@
...
+    return 6
+  }
+  else if (compileVersion == 33) {
+    return 5
+  }
+  else if (compileVersion == 32) {
+    return 4
+  }
+  else {
+    throw new GradleException("The version of Capacitor is not supported by Sentry, CODE: " + compileVersion)
```
Comment (krystofwoldrich) [inline android/build.gradle:23]: This seems like a breaking change.

Did the SDK build with Capacitor 3 before?

File: example/ionic-angular-v3/scripts/skip-pod-windows.js
Comment (krystofwoldrich) [inline example/ionic-angular-v3/scripts/skip-pod-windows.js:1]: This script is not used. Probably duplicate from the top level scripts.

File: android/build.gradle (hunk 1 above)
Comment (krystofwoldrich) [reply android/build.gradle:23]: Or does the case for 4 covers 3 as well?

**Comment by krystofwoldrich** [comment]
The SDK is in v0.X.X so it's okay to drop some very old capacitor version, but we should mentioned in Changelog.

I'm not sure if we dropped any, from the code it looks like maybe 2 and 3?

File: android/build.gradle (hunk 1 above)
Comment (lucas-zimerman) [reply android/build.gradle:23]: Nice catch, I'll fix the case for Capacitor 3

File: package.json [hunk 2]
```diff
@@ -70,9 +70,9 @@
     "@sentry/utils": "7.114.0"
   },
   "devDependencies": {
-    "@capacitor/android": "^3.0.1 || ^4.0.0 || ^5.0.0",
-    "@capacitor/core": "^3.0.1 || ^4.0.0 || ^5.0.0",
-    "@capacitor/ios": "^3.0.1 || ^4.0.0 || ^5.0.0",
+    "@capacitor/android": "^3.0.1 || ^4.0.0 || ^5.0.0 || ^6.0.0",
+    "@capacitor/core": "^3.0.1 || ^4.0.0 || ^5.0.0 || ^6.0.0",
+    "@capacitor/ios": "^3.0.1 || ^4.0.0 || ^5.0.0 || ^6.0.0",
```
[the remark this answers is not in the document]
Comment (lucas-zimerman) [reply package.json:75]: it's valid

**Comment by jcesarmobile** [comment]
Capacitor 4 and older are out of support, I don’t think you should support those versions anymore

https://capacitorjs.com/docs/main/reference/support-policy

File: android/build.gradle [hunk 3]
```diff
@@ -1,60 +1,58 @@
...
-    if (rootProject.project(':capacitor-android').android.compileOptions.sourceCompatibility >= JavaVersion.VERSION_17) {
-      println "Sentry Capacitor: Applying dependencies for Capacitor 5 or higher."
-      classpath 'com.android.tools.build:gradle:8.0.0'
-      classpath 'com.google.gms:google-services:4.3.15'
-    }
-    else {
-      println "Sentry Capacitor: Applying dependencies for Capacitor 4."
-      classpath 'com.android.tools.build:gradle:7.2.1'
-      classpath 'com.google.gms:google-services:4.3.13'
-    }
```
[the remark this answers is not in the document]
Comment (lucas-zimerman) [reply android/build.gradle:43]: the SDK doesn't depend on google services and the gradle version is going to be used from the parent project. Making this code unnecessary and harder to maintain once new versions arrive.

[the remark this answers is not in the document]
Comment (krystofwoldrich) [reply android/build.gradle:43]: Awesome, thanks for the explanation.

**Review by krystofwoldrich — APPROVED** [review]
Looks good, thank you! 🚀

**Pull request merged**